crepecake/src/main/java/com/google/cloud/tools/crepecake/image/LayerBuilder.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.crepecake.image;
+
+import com.google.cloud.tools.crepecake.tar.TarStreamBuilder;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+
+/** Builds an {@link UnwrittenLayer} from files. */
+public class LayerBuilder {
+
+  /** The partial filesystem changeset to build the layer with. */
+  private final List<TarArchiveEntry> filesystemEntries = new ArrayList<>();
+
+  /**
+   * Prepares a file to be built into the layer.
+   *
+   * @param file the file to add
+   * @param path the path of the file in the partial filesystem changeset
+   */
+  public void addFile(File file, String path) {
```

Comment:
Noticed this in the test. Are these really two independent arguments? Or is the path always formed from the file in an obvious way?